### PR TITLE
feat: optimize actions

### PR DESCRIPTION
**Pull request by devhaozi**
## 📑 Description

Closes https://github.com/goravel/goravel/issues/?

## Summary by CodeRabbit

- **New Features**
	- Introduced caching for Go modules in multiple workflows, enhancing build efficiency.
  
- **Bug Fixes**
	- Streamlined caching configurations to improve performance and reduce complexity in CI/CD processes.

- **Refactor**
	- Updated linting and testing workflows to utilize built-in caching capabilities, ensuring core functionalities remain intact while optimizing performance.

## ✅ Checks

- [x] Added test cases for my code

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -10,17 +10,8 @@ jobs:
...
-          cache: false
-      - name: Go mod cache
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
```
Comment (devhaozi) [inline .github/workflows/lint.yml:23]: 不需要手动添加缓存，setup-go会自动应用缓存。

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -36,18 +27,10 @@ jobs:
...
           go-version: "stable"
-      - name: Go mod cache
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
```
Comment (devhaozi) [inline .github/workflows/lint.yml:47]: 同上

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -12,15 +12,7 @@ jobs:
...
           go-version: ${{ matrix.go }}
-      - name: Go mod cache
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
```
Comment (devhaozi) [inline .github/workflows/test.yml:23]: 同上

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -35,16 +27,8 @@ jobs:
...
           go-version: ${{ matrix.go }}
-      - name: Go mod cache
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~\AppData\Local\go-build
-            ~\go\pkg\mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
```
Comment (devhaozi) [inline .github/workflows/test.yml:46]: 同上

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -36,18 +27,10 @@ jobs:
...
+          cache: true
       - name: Install dependencies
         run: go mod tidy
       - name: Install Nilaway
         run: go install go.uber.org/nilaway/cmd/nilaway@latest
       - name: Run Nilaway
-        run: nilaway ./...
```
Comment (devhaozi) [inline .github/workflows/lint.yml:53]: nilaway只检查goravel的包，不检查第三方包和标准库。

**Review by hwbrzzl — APPROVED** [review]

**Comment by hwbrzzl** [comment]
Good catch 👍

**Pull request merged**